src/index.ts

```diff
@@ -170,19 +170,17 @@ export const htmlPlugin = (configuration: Configuration = { files: [], }): esbui
         return `${publicPath}${slash}${relPath}`
     }
 
-    async function injectFiles(dom: JSDOM, assets: { path: string }[], outDir: string, publicPath: string | undefined, htmlFileConfiguration: HtmlFileConfiguration) {
-        const document = dom.window.document
+    async function injectFiles(dom: cheerio.CheerioAPI, assets: { path: string }[], outDir: string, publicPath: string | undefined, htmlFileConfiguration: HtmlFileConfiguration) {
         for (const script of htmlFileConfiguration?.extraScripts || []) {
-            const scriptTag = document.createElement('script')
+            const scriptTag = dom('body').append('<script />')
```

Comment:
I'm not the biggest fan of the name `dom` in this context - would you mind renaming it back to document?